go/vt/callerid/callerid.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package callerid stores/retrives CallerIDs (immediate CallerID
+// and effective CallerID) to/from the Context
+package callerid
+
+import (
+	"golang.org/x/net/context"
+)
+
+// ImmediateCallerID is the username of immediate caller to VTGate
+type ImmediateCallerID struct {
+	username string
+}
```

Comment:
I would just use the proto3.VTGateCallerID structure for this. No need to have an internal structure and copy stuff around for proto3, which is the direction we're heading.